x-pack/plugins/secops/server/lib/events/elasticsearch_adapter.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, has, isEmpty, merge } from 'lodash/fp';
+import { EventItem, EventsData, KpiItem } from '../../../common/graphql/types';
+import { FrameworkAdapter, FrameworkRequest } from '../framework';
+import {
+  EventData,
+  EventFilterQuery,
+  EventsAdapter,
+  EventsRequestOptions,
+  TermAggregation,
+} from './types';
+
+const EventFieldsMap = {
+  timestamp: '@timestamp',
+  'host.hostName': 'host.name',
+  'host.ip': 'host.ip',
+  'event.type': 'event.type',
+  'event.severity': 'event.severity',
+  'suricata.flow_id': 'suricata.eve.flow_id',
+  'suricata.proto': 'suricata.eve.proto',
+  'source.ip': 'source_ecs.ip',
+  'source.port': 'source_ecs.port',
+  'destination.ip': 'destination.ip',
+  'destination.port': 'destination.port',
+  'geo.region_name': 'destination.geo.country_iso_code',
+  'geo.country_iso_code': 'destination.geo.region_name',
+};
+
+export class ElasticsearchEventsAdapter implements EventsAdapter {
+  private framework: FrameworkAdapter;
+  constructor(framework: FrameworkAdapter) {
+    this.framework = framework;
+  }
+
+  public async getEvents(
+    request: FrameworkRequest,
+    options: EventsRequestOptions
+  ): Promise<EventsData> {
+    const { to, from } = options.timerange;
+    const Fields = options.fields;
+    const filterQuery = options.filterQuery;
+    const EsFields = Fields.reduce(
+      (res, f: string) => {
+        if (EventFieldsMap.hasOwnProperty(f)) {
+          const esField = Object.getOwnPropertyDescriptor(EventFieldsMap, f);
+          if (esField && esField.value) {
+            res.push(esField.value);
```

Comment:
`nit`: consider avoiding the array mutation here using `[...spread]`, or slightly less preferably, `concat`.

This short video by Dan Abramov (author of Redux, and a member of the React team) is a good resource:  https://egghead.io/lessons/react-redux-avoiding-array-mutations-with-concat-slice-and-spread

Reply:
👍